### PR TITLE
Fix method data fetcher

### PR DESCRIPTION
**Pull request by yarinvak**
fix https://github.com/graphql-java/graphql-java-annotations/issues/174

### CODE REVIEW

File: src/main/java/graphql/annotations/dataFetchers/MethodDataFetcher.java [hunk 1]
```diff
@@ -135,9 +136,51 @@ private Object buildArg(Type p, GraphQLType graphQLType, Object arg) {
         }
     }
 
-    private Object getFieldValue(Object source, String fieldName) throws IllegalAccessException, NoSuchFieldException {
-        Field field = source.getClass().getDeclaredField(fieldName);
-        field.setAccessible(true);
-        return field.get(source);
+    private Object getGraphQLFieldValue(Object source, String fieldName) throws IllegalAccessException, NoSuchFieldException, InvocationTargetException {
```
Comment (guy120494) [inline src/main/java/graphql/annotations/dataFetchers/MethodDataFetcher.java:139]: Separate to few methods, one that deals with fields and one that deals with methods

File: src/main/java/graphql/annotations/dataFetchers/MethodDataFetcher.java [hunk 2]
```diff
@@ -135,9 +136,51 @@ private Object buildArg(Type p, GraphQLType graphQLType, Object arg) {
...
+        Method method = getMethod(source.getClass(), fieldName, "");
+        Method getMethod = getMethod(source.getClass(), fieldName, "get");
```
Comment (guy120494) [inline src/main/java/graphql/annotations/dataFetchers/MethodDataFetcher.java]: call this method only if necessary (reflection takes a lot of time)

File: src/main/java/graphql/annotations/dataFetchers/MethodDataFetcher.java [hunk 3]
```diff
@@ -135,9 +136,51 @@ private Object buildArg(Type p, GraphQLType graphQLType, Object arg) {
...
+        Method isMethod = getMethod(source.getClass(), fieldName, "is");
```
Comment (guy120494) [inline src/main/java/graphql/annotations/dataFetchers/MethodDataFetcher.java]: call this method only if necessary (reflection takes a lot of time)

File: src/main/java/graphql/annotations/processor/util/PrefixesUtil.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ */
+package graphql.annotations.processor.util;
+
+public class PrefixesUtil {
+    public static String createPrefix(String prefix, String propertyName) {
```
Comment (guy120494) [inline src/main/java/graphql/annotations/processor/util/PrefixesUtil.java]: The name of the function doesn't explain what the function does

File: src/main/java/graphql/annotations/dataFetchers/MethodDataFetcher.java [hunk 5]
```diff
@@ -135,9 +136,51 @@ private Object buildArg(Type p, GraphQLType graphQLType, Object arg) {
...
+        }
+
+        return method;
     }
+
+    private Field getField(Class<?> clazz, String name) {
+        Field field = null;
+        while (clazz != null && field == null) {
+            try {
+                field = clazz.getDeclaredField(name);
```
Comment (guy120494) [inline src/main/java/graphql/annotations/dataFetchers/MethodDataFetcher.java:186]: should it be "getDeclaredField" or "getField"?

**Review by guy120494 — CHANGES_REQUESTED** [review]

File: src/main/java/graphql/annotations/dataFetchers/MethodDataFetcher.java (hunk 5 above)
Comment (yarinvak) [reply src/main/java/graphql/annotations/dataFetchers/MethodDataFetcher.java:186]: getField returns only public fields (but for entire hirearchy), but we need private ones as well. So this method runs over the entire hirearchy and searches for private methods with this name.

**Pull request merged**